platform/macos/display_server_macos.mm

```diff
@@ -2454,7 +2467,7 @@
 					wd.layered_window = false;
 					set_window_per_pixel_transparency_enabled(false, p_window);
 				}
-				[wd.window_object setStyleMask:NSWindowStyleMaskTitled | NSWindowStyleMaskClosable | NSWindowStyleMaskMiniaturizable | (wd.extend_to_title ? NSWindowStyleMaskFullSizeContentView : 0) | (wd.resize_disabled ? 0 : NSWindowStyleMaskResizable)];
+				[wd.window_object setStyleMask:NSWindowStyleMaskTitled | NSWindowStyleMaskClosable | (wd.minimize_disabled ? 0 : NSWindowStyleMaskMiniaturizable) | (wd.extend_to_title ? NSWindowStyleMaskFullSizeContentView : 0) | (wd.resize_disabled ? 0 : NSWindowStyleMaskResizable)];
```

Comment:
Full-screen transitions in the `godot_window_delegate.mm` have the same code and should be undated as well, see `windowWillEnterFullScreen:`, and `windowDidExitFullScreen:` methods.

And also in the `case WINDOW_FLAG_TRANSPARENT:` below.